src/app/services/stack.service.ts

```diff
@@ -0,0 +1,65 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Card } from '../models/card';
+import { Stack } from '../models/stack';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class StackService {
+
+  constructor(private http:HttpClient) { }
+
+  // CRUD Operations
+  async createStack(stack:Stack):Promise<Stack>{
+    stack = await this.http.post<Stack>("http://34.122.220.146:8080/stacks",stack).toPromise()
+    return stack;
+  }
+
+  async getAllStacks():Promise<Stack[]>{
+    const stacks:Stack[] = await this.http.get<Stack[]>(`http://34.122.220.146:8080/stacks`).toPromise();
+    return stacks;
+  }
+
+  async getStackById(stackId:number):Promise<Stack>{
+    const stack:Stack = await this.http.get<Stack>(`http://34.122.220.146:8080/stacks/${stackId}`).toPromise();
+    return stack;
+  }
+
+  async updateStack(stack:Stack):Promise<Stack>{
+    const updatedStack:Stack = await this.http.put<Stack>(`http://34.122.220.146:8080/stacks/${stack.stackId}`,stack).toPromise();
+    return stack;
+  }
+
+  async deleteStack(stackId:number):Promise<Object>{
+    const result = await this.http.delete(`http://34.122.220.146:8080/stacks/${stackId}`).toPromise();
+    return result;
+  }
+
+  // Services for Cards in Stacks
+  async getAllCardsFromStack(stackId:number){
```

Comment:
The Stack object itself should contain an array that has the related cards. This may be a bit redundant. We should communicate some more about the implementation of stack => cards